### ISSUE TITLE
[java] UnusedFormalParameter doesn't consider anonymous classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3618]
**Affects PMD Version:** 6.40.0

**Rule:** [UnusedFormalParameter](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedformalparameter)

**Description:**

Methods in anonymous classes are not considered.

**Code Sample demonstrating the issue:**

```java
class Impl {
    public void m4() {
        Object o = new Object() {
            private void m5(int arg1) { } // unsed formal parameter
            public boolean equals(Object other) { m5(0); return false; }
        };
        o.equals(o);
    }
}

```

**Expected outcome:**

PMD should report a violation at line 4, but doesn't. This is a false-negative.
